src/metabase/api/card.clj

```diff
@@ -601,16 +601,16 @@ saved later when it is ready."
 (defn- publish-card-update!
   "Publish an event appropriate for the update(s) done to this CARD (`:card-update`, or archiving/unarchiving
   events)."
-  [card archived?]
+  [card-before-updates card-after-update archived?]
   (let [event (cond
                 ;; card was archived
                 (and archived?
-                     (not (:archived card))) :card-archive
+                     (not (:archived card-before-updates))) :card-archive
```

Comment:
I feel like this is more clearly expressed without using the `archived?` arg. You should also be able to get rid of the comments like
```clojure
(and (:archived card-after-updates)
     (not (:archived card-before-updates))) :card-archive
```

Reply:
it's better! I made the change in 88da822e5d